crates/subspace-service/src/sync_from_dsn/import_blocks.rs

```diff
@@ -238,8 +256,6 @@ where
             // Import queue handles verification and importing it into the client
             import_queue_service.import_blocks(BlockOrigin::NetworkInitialSync, blocks_to_import);
         }
-
-        *last_processed_segment_index = segment_index;
```

Comment:
I think the problem is in this line in the `main` branch:
* when we update the segment index above, we check we've fully processed all blocks from that segment
* but when we update it here, we don't check for a partial block at the end of the segment


```rust
        // Segments are only fully processed when all their blocks are processed.
        if last_archived_block_partial {
            *last_processed_segment_index = segment_index.saturating_sub(1);
        } else {
           *last_processed_segment_index = segment_index;
        } 
```